01-data-structures/02-stacks-and-queues/myqueue/myqueue.rb

```diff
@@ -5,14 +5,25 @@ class MyQueue
   def initialize
     @queue = Array.new
     @head = @queue[0]
+		@tail = @queue[-1] # added by Tom
   end
 
   def enqueue(element)
+		@queue.insert(0, element)
+		@head = @queue[0]
+		@tail = @queue[-1]
   end
 
   def dequeue
+		@queue.delete(@queue[-1])
```

Comment:
There is actually a `delete_at` method on arrays, so this could be `@queue.delete_at(-1)`.